### PR TITLE
[SPARK-48201][DOCS][PYTHON] Make some corrections in the docstring of pyspark DataStreamReader methods

### PR DESCRIPTION
**Pull request by chloeh13q**
### What changes were proposed in this pull request?

The docstrings of the pyspark DataStream Reader methods `csv()` and `text()` say that the `path` parameter can be a list, but actually when a list is passed an error is raised.

### Why are the changes needed?

Documentation is wrong.

### Does this PR introduce _any_ user-facing change?

Yes. Fixes documentation.

### How was this patch tested?

N/A

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Review by flaviaouyang — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Thanks for the fix. Mind filing a JIRA please? see also https://spark.apache.org/contributing.html

**Comment by chloeh13q** [comment]
@HyukjinKwon Yep! Here it is: https://issues.apache.org/jira/browse/SPARK-48201

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
LGTM once CI passes thanks!

**Comment by xinrong-meng** [comment]
Let's add `[DOCS]` to the pr title please

File: python/pyspark/sql/streaming/readwriter.py [hunk 1]
```diff
@@ -641,8 +641,8 @@ def csv(
 
         Parameters
         ----------
-        path : str or list
```
Comment (allisonwang-db) [inline python/pyspark/sql/streaming/readwriter.py:644]: Just curious, what's the error message if you pass a list here?

Comment (flaviaouyang) [reply python/pyspark/sql/streaming/readwriter.py:644]: It will throw a PySparkTypeError saying argument passed is not string

Comment (chloeh13q) [reply python/pyspark/sql/streaming/readwriter.py:644]: Yes, it explicitly checks if the argument is a string:
https://github.com/apache/spark/blob/master/python/pyspark/sql/streaming/readwriter.py#L717-L723

Comment (allisonwang-db) [reply python/pyspark/sql/streaming/readwriter.py:644]: Got it thanks!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @chloeh13q and all.
Merged to master for Apache Spark 4.0.0.

**Comment by dongjoon-hyun** [comment]
Welcome to the Apache Spark community, @chloeh13q . 
I added you to the Apache Spark contributor group and assigned SPARK-48201 to you.
Congratulations for your first commit!

**Pull request closed without merging**